### PR TITLE
Build: tackle Gradle deprecations marked as for-removal in Gradle 9

### PR DESCRIPTION
**Pull request by snazy**

### CODE REVIEW

File: build-logic/build.gradle.kts [hunk 1]
```diff
@@ -45,3 +45,7 @@ tasks.register("codeChecks").configure {
   description = "Runs code style and license checks"
   dependsOn("spotlessCheck")
 }
+
+kotlin {
```
Comment (snazy) [inline build-logic/build.gradle.kts:49]: Gradle complains about Java 23 vs Kotlin targeting 22 max (and going to fail in Gradle 9 in such a case).

File: build-logic/src/main/kotlin/Utilities.kt [hunk 2]
```diff
@@ -214,9 +215,9 @@ fun DependencyHandlerScope.nessieProject(
   if (artifactId.startsWith(":")) {
     throw IllegalArgumentException("artifactId for nessieProject() must not start with ':'")
   }
-  return if (!isIncludedInNesQuEIT(project(":").dependencyProject.gradle)) {
```
Comment (snazy) [inline build-logic/src/main/kotlin/Utilities.kt:217]: That `dependencyProject` getter is deprecated for removal in Gradle 9

File: build-logic/src/main/kotlin/nessie-common-src.gradle.kts [hunk 3]
```diff
@@ -178,7 +178,7 @@ class MemoizedGitInfo {
   companion object {
     private fun execProc(rootProject: Project, cmd: String, vararg args: Any): String {
       val buf = ByteArrayOutputStream()
-      rootProject.exec {
+      rootProject.providers.exec {
```
Comment (snazy) [inline build-logic/src/main/kotlin/nessie-common-src.gradle.kts:181]: Not sure why, but `Project.exec` got deprecated for removal in 9, but doing that via `ProviderFactory.exec` is not :shrug:

File: tools/doc-generator/site-gen/build.gradle.kts [hunk 4]
```diff
@@ -166,99 +166,87 @@ val cliHelp by tasks.registering(JavaExec::class) {
...
-  mainClass = gcMainClass
-  args("--help")
-
-  doFirst {
-    delete(gcHelpDir)
-  }
-
-  standardInput = InputStream.nullInputStream()
-  standardOutput = ByteArrayOutputStream()
+val gcHelp by tasks.registering(Sync::class) {
```
Comment (snazy) [inline tools/doc-generator/site-gen/build.gradle.kts:169]: The change in this project is to generate each help output using one Gradle task per invocation into distinct directories and then collect those using a `Sync` task

... `javaexec` is deprecated for removal in 9

**Comment by snazy** [comment]
CI failure is unrelated, should be fixed via #9970

**Review by dimas-b — APPROVED** [review]

**Pull request merged**